### PR TITLE
docs: upgrade docs now reflect the new upgrade commands

### PR DESCRIPTION
**Pull request by derpsteb**
### Proposed change(s)

- The commands have changed greatly, thus I replaced the larger part of the upgrade docs.
- I also replaced the term `component(s)` with `microservice(s)` when referring to our microservices. Since users are now faced with a config value called `microserviceVersion`, I feel like something should come up when searching the docs for "microservice".

### Checklist

- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone

### CODE REVIEW

File: docs/docs/architecture/microservices.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
-# Components
+# Microservices
```
Comment (thomasten) [inline docs/docs/architecture/microservices.md:1]: filename must also be renamed in sidebars.js

File: docs/docs/workflows/upgrade.md [hunk 2]
```diff
@@ -1,44 +1,50 @@
 # Upgrade your cluster
 
-Constellation provides an easy way to upgrade to the next release.
-This involves updating the CLI, choosing a new VM image to use for all nodes in the cluster, and updating the cluster's expected measurements.
+Constellation provides an easy way to upgrade all components of your cluster, without disrupting it's availability.
+Specifically you can upgrade the Kubernetes version, the nodes' image and the Constellation microservices.
```
Comment (thomasten) [inline docs/docs/workflows/upgrade.md:4]:
```suggestion
Constellation provides an easy way to upgrade all components of your cluster, without disrupting its availability.
Specifically, you can upgrade the Kubernetes version, the nodes' image, and the Constellation microservices.
```
MS style guide recommends a comma following introductory phrases and also the Oxford comma

File: docs/docs/workflows/upgrade.md [hunk 3]
```diff
@@ -1,44 +1,50 @@
...
 ## Update the CLI
 
-New features and bug fixes are added to the CLI with every release. To use them, update the CLI to the latest version by following the instructions in the [installation guide](../getting-started/install.md).
+Each CLI comes with a set of microservice and Kubernetes versions
+It's recommend to always use the most up-to-date CLI.
+However, there are two constraints to this recommendation:
+1. Don't upgrade more than one minor version at a time.
+For example, if you are currently on CLI version v2.6, you should upgrade to v2.7, upgrade your cluster and only then continue upgrading the CLI after.
+2. Don't upgrade to a CLI version that doesn't support your current Kubernetes version.
+If your current Kubernetes version is dropped from the next CLI version, use your current CLI to upgrade to a newer Kubernetes version first.
```
Comment (thomasten) [inline docs/docs/workflows/upgrade.md:17]: I find this hard to understand. I tried rephrasing it to what you should do instead of what you shouldn't do:
```suggestion
Each CLI comes with a set of supported microservice and Kubernetes versions.
Most importantly, a given CLI version can only upgrade a cluster of the previous minor version, but not older ones.
This means that you have to upgrade your CLI and cluster one minor version at a time.

For example, if you are currently on CLI version v2.6 and the latest one is v2.8, you should

* upgrade the CLI to v2.7,
* upgrade the cluster to v2.7,
* and only then continue upgrading the CLI (and the cluster) to v2.8 after.

Also note that if your current Kubernetes version isn't supported anymore by the next CLI version, use your current CLI to upgrade to a newer Kubernetes version first.
```
What do you think?

File: docs/docs/workflows/upgrade.md [hunk 4]
```diff
@@ -1,44 +1,50 @@
...
+Each CLI supports at least three Kubernetes versions.
+To learn which Kubernetes versions are supported by a particular CLI run [constellation config kubernetes-versions](../reference/cli.md#constellation-config-kubernetes-versions).
```
Comment (thomasten) [inline docs/docs/workflows/upgrade.md:19]:
```suggestion
To learn which Kubernetes versions are supported by a particular CLI, run [constellation config kubernetes-versions](../reference/cli.md#constellation-config-kubernetes-versions).
```

File: docs/docs/workflows/upgrade.md [hunk 5]
```diff
@@ -1,44 +1,50 @@
...
 ## Migrate the configuration
 
 The Constellation configuration file is located in the file `constellation-conf.yaml` in your workspace.
 Refer to the [migration reference](../reference/config-migration.md) to check if you need to update fields in your configuration file.
 
-## Plan the upgrade
+## Check for upgrades
 
-If you don't already know the image you want to upgrade to, use the `upgrade plan` command to pull a list of available updates.
+To learn which versions the current CLI can upgrade to and what's installed in your cluster use this command to see available upgrades:
```
Comment (thomasten) [inline docs/docs/workflows/upgrade.md:29]:
```suggestion
To learn which versions the current CLI can upgrade to and what's installed in your cluster, run:
```

File: docs/docs/workflows/upgrade.md [hunk 6]
```diff
@@ -1,44 +1,50 @@
...
-
-:::caution
-
-`constellation upgrade plan` only works for official Edgeless release images.
-If your cluster is using a custom image, the Constellation CLI will fail to find compatible images.
-However, you may still use the `upgrade execute` command by manually selecting a compatible image and setting it in your config file.
-
-:::
+You can either enter the reported target versions into your config manually or run the above command with the `--write-config` flag.
+When using this flag the `kubernetesVersion`, `image` and `microserviceVersion` fields are overwritten with the smallest available upgrade.
```
Comment (thomasten) [inline docs/docs/workflows/upgrade.md:36]:
```suggestion
When using this flag, the `kubernetesVersion`, `image`, and `microserviceVersion` fields are overwritten with the smallest available upgrade.
```

File: docs/docs/workflows/upgrade.md [hunk 7]
```diff
@@ -1,44 +1,50 @@
...
 
-## Execute the upgrade
+## Apply the upgrade
 
-Once your config file has been prepared with the new image and measurements, use the `upgrade execute` command to initiate the upgrade.
+Once you updated your config with the versions you want in your cluster, you can trigger the upgrade with this command:
```
Comment (thomasten) [inline docs/docs/workflows/upgrade.md:40]:
```suggestion
Once you updated your config with the desired versions, you can trigger the upgrade with this command:
```

File: docs/docs/workflows/upgrade.md [hunk 8]
```diff
@@ -1,44 +1,50 @@
...
-constellation upgrade execute
+constellation upgrade apply
 ```
 
-After the command has finished, the cluster will automatically replace old nodes using a rolling update strategy to ensure no downtime of the control or data plane.
+Microservice upgrades will be finished within a few minutes, depending on the cluster size.
+If you are interested, you can monitor pods restarting in the `kube-system` namespace with your tool of choice.
+Image and Kubernetes upgrades will take longer.
+This is because for each node in your cluster, a new node has to be created and joined.
+The process usually takes ~6 minutes per node.
```
Comment (thomasten) [inline docs/docs/workflows/upgrade.md:50]:
```suggestion
If you are interested, you can monitor pods restarting in the `kube-system` namespace with your tool of choice.

Image and Kubernetes upgrades take longer.
For each node in your cluster, a new node has to be created and joined.
The process usually takes about 6 minutes per node.
```
I'd start a new paragraph (and simplify the wording a bit).

6 minutes sound like a very accurate timing. Is this really true, independent of CSP or node type? (If not, maybe replace with sth like "about five minutes" or "up to ten minutes"?)

File: docs/docs/workflows/upgrade.md (hunk 3 above)
Comment (derpsteb) [reply docs/docs/workflows/upgrade.md:17]: I like it a lot. Changed it.

**Review by thomasten — APPROVED** [review]
Thanks, LGTM! I'm still unsure if upgrades are sufficiently documented. But I can't tell anything specifically missing. We may learn from our users sometime ;)
